internal/k8s/reconciler/gateway.go

```diff
@@ -48,40 +46,8 @@ type K8sGatewayConfig struct {
 	Client          gatewayclient.Client
 }
 
-func NewK8sGateway(gateway *gw.Gateway, config K8sGatewayConfig) *K8sGateway {
-	gatewayLogger := config.Logger.Named("gateway").With("name", gateway.Name, "namespace", gateway.Namespace)
-	listeners := make(map[string]*K8sListener)
-	for _, listener := range gateway.Spec.Listeners {
-		k8sListener := NewK8sListener(gateway, listener, K8sListenerConfig{
-			ConsulNamespace: config.ConsulNamespace,
-			Logger:          gatewayLogger,
-			Client:          config.Client,
-		})
-		listeners[k8sListener.ID()] = k8sListener
-	}
-
-	deployment := builder.NewGatewayDeployment(gateway)
-	deployment.WithSDS(config.SDSHost, config.SDSPort)
-	deployment.WithClassConfig(config.Config)
-	deployment.WithConsulCA(config.ConsulCA)
-	deployment.WithConsulGatewayNamespace(config.ConsulNamespace)
-	service := builder.NewGatewayService(gateway)
-	service.WithClassConfig(config.Config)
-
-	return &K8sGateway{
-		config:            config.Config,
-		deploymentBuilder: deployment,
-		serviceBuilder:    service,
-		consulNamespace:   config.ConsulNamespace,
-		logger:            gatewayLogger,
-		client:            config.Client,
-		gateway:           gateway,
-		listeners:         listeners,
-	}
-}
-
 func (g *K8sGateway) Validate(ctx context.Context) error {
```

Comment:
Most of the logic from here down will eventually get moved into a dedicated `GatewayValidator` in a subsequent PR.